mlir/utils/performance/quickTunerStat.py

```diff
@@ -0,0 +1,565 @@
+        """
+        Method to validate a set of .qt files
+        """
+        raise NotImplementedError()
+            
+
+class DataValidator(PerfConfigValidator):
+    """
+    uses already provided data to validate the configs generated
+    """
+    def __init__(self, gemm_config_file, preproc_file, debug=False):
+        super().__init__()
+        self.gemm_configs = super().collectGemmConfigs(gemm_config_file) # list of GemmConfiguratioin
+        self.gemm_keys = [(cfg.transA, cfg.transB, cfg.g, cfg.m, cfg.n, cfg.k) for cfg in self.gemm_configs]
+        #self.gemm_keys = [super(DataValidator, self).gemmConfigToKey(gemm) for gemm in self.gemm_configs]
+        self.preproc_file = preproc_file
+        self.validation_data = super().orderByGemmType(self.preproc_file)
+        self.debug = debug
+
+    def __typeQtMap(self, input_dir):
+        """
+        creates a dictionary keyed by qt type with values being
+        list of matching qt files
+        """        
+        qt_files = glob.glob(os.path.join(input_dir, "*.qt"))
+        file_dict = {}
+
+        for file in qt_files:
+            file_split = os.path.basename(file).split('.')
+            file_type = file_split[1]
+            method = file_split[0]
+            
+            if method not in file_dict:
+                file_dict[method] = {}
+            file_dict[method][file_type] = pd.read_csv(file)
+            
+        return file_dict
+
+    def validateDir(self, input_dir):
+        """
+        process whole directory
+        """
+        self.quick_tune_data = self.__typeQtMap(input_dir)
+        
+        output_dict = {}
+        for method in self.quick_tune_data:
+            if self.debug:
+                print(f"method {method}")
+            for dtype in self.quick_tune_data[method]:
+                if self.debug:
+                    print(f"dtype {dtype}")
+                if dtype not in self.validation_data:
+                    continue
+                if dtype not in output_dict:
+                    output_dict[dtype] = {}
+                if self.debug:
+                    print(f"quick_tune_data {self.quick_tune_data[method][dtype]}")    
+                gemm_data = self.validate(self.quick_tune_data[method][dtype], dtype)
+                output_dict[dtype][method] = gemm_data
+        self.output_dict = output_dict
+        return output_dict
+        
+    def validateFile(self, input_file, dtype=None):
+        """
+        process single file, if type passed in then we read perf config file
+        """
+        if not dtype:
+            dtype = os.path.basename(input_file).split('.')[1]
+        df = super().readPerfConfig(input_file)
+        return self.validate(df, dtype)        
+    
+    def validate(self, file_data, dtype):
+        # to validate file we need data already read,
+
+        all_data = []
+        columns = ['M/block', 'N/block', 'K/block', 'M/wave', 'N/wave', 'kPack', 'splitK', 'forceUnroll', 'bCopyMore']
```

Comment:
I'll note that, for example, the attention perf configs don't have this structure, and, more interestingl, that f32 gemm/conv on Navi doesn't either. But this is probably fine, just wanted to call out the limitation